argo-cd-apps/base/host/ingresscontroller/ingresscontroller.yaml

```diff
@@ -10,17 +10,22 @@ spec:
         generators:
           - clusters:
               values:
-                environment: ""
-                clusterDir: ""
+                sourceRoot: configs/ingresscontroller
+                environment: staging
```

Comment:
we are missing somthing that will replace environment value from staging to production in the production overlay

Reply:
should be fixed now, added a patch for this in its overlay.